--- drakx-kbd-mouse-x11-0.102/Makefile.orig	2013-10-29 23:01:37.691586259 +0000
+++ drakx-kbd-mouse-x11-0.102/Makefile	2013-10-29 23:02:11.274588026 +0000
@@ -12,7 +12,7 @@
 pixmapsdir=/usr/share/libDrakX/pixmaps
 
 BIN_TOOLS= 
-SBIN_TOOLS= keyboarddrake mousedrake XFdrake
+SBIN_TOOLS= keyboarddrake mousedrake XFdrake drakx-update-background
 ROOTSBIN_TOOLS = display_driver_helper
 INLIBDEST_DIRS = lib/xf86misc
 
