From f2a1da8b2f1d503b7c7fa3fc7e1e6d5d3a7a1973 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tomasz=20Pawe=C5=82=20Gajc?= <phenomenal@wp.pl>
Date: Tue, 29 Oct 2013 13:47:16 +0400
Subject: [PATCH] fix path to dumpkeys

---
 lib/keyboard.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/keyboard.pm b/lib/keyboard.pm
index 64bd108..5b1ec38 100644
--- a/lib/keyboard.pm
+++ b/lib/keyboard.pm
@@ -633,7 +633,7 @@ sub write {
             $_ .= $s if eof;
         } "$::prefix/etc/sysctl.conf";
     } else {
-	run_program::rooted($::prefix, 'dumpkeys', '>', '/etc/sysconfig/console/default.kmap') or log::l("dumpkeys failed");
+	run_program::rooted($::prefix, '/bin/dumpkeys', '>', '/etc/sysconfig/console/default.kmap') or log::l("dumpkeys failed");
     }
     if (-x $localectl) { # systemd service
 	run_program::run($localectl, '--no-convert', 'set-keymap', 
-- 
1.7.11.3

